TypeSystem/flow-analysis/reachability_switch_A08_t02.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion switch statement: If `N` is a switch statement of the form
+/// `switch (E) {alternatives}` then:
+/// - Let `before(E) = before(N)`.
+/// - For each `C` in `alternatives` with statement body `S`:
+///   - If `C` is labelled let
+///     `before(S) = conservativeJoin(after(E), assignedIn(N), capturedIn(N))`
+///     otherwise let `before(S) = after(E)`.
+///  - If the cases are exhaustive, then let `after(N) = break(N)` otherwise let
+///    `after(N) = join(after(E), break(N))`.
+///
+/// @description Checks that if a variable is initialized in a `when` part of a
+/// `case` statement then this variable is not definitely assigned.
```

Comment:
If you say `int _` then it seems likely that the flow analysis _could_ know that the case will definitely match and hence the `when` expressions will definitely be evaluated (and it may be true or false, but that doesn't change the fact that the `when` expression has been evaluated).

So maybe this test will fail, and we should expect `i` to be definitely assigned after all? Let's rely on the implementation behavior in order to see which way it goes.

Reply:
Ok, the next test already does this. Closing.